pkg/commands/start_test.go

```diff
@@ -16,67 +16,84 @@ package commands
 import (
 	"context"
 	"fmt"
+	"math/rand"
 	"net"
+	"os"
 	"os/exec"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/require"
 )
 
-var (
-	tcpPort = "5353"
-	udpPort = "5353"
-)
-
 func TestStart(t *testing.T) {
-	cmd := run(t, "127.0.0.1", tcpPort, udpPort, "host.rd.test=111.111.111.111,host2.rd.test=222.222.222.222")
+	port := randomPort()
+	cmd := runHostResovler(t, []string{"-a", "127.0.0.1", "-t", port, "-u", port})
 	defer cmd.Process.Kill()
 
-	t.Logf("Checking for TCP port is running on %v", tcpPort)
-	tcpListener, err := net.Listen("tcp", fmt.Sprintf(":%s", tcpPort))
+	t.Logf("Checking for TCP port is running on %v", port)
+	tcpListener, err := net.Listen("tcp", fmt.Sprintf(":%s", port))
 	if tcpListener != nil {
 		defer tcpListener.Close()
 	}
-	require.Errorf(t, err, "host-resolver is not listening on TCP port %s", tcpPort)
+	require.Errorf(t, err, "host-resolver is not listening on TCP port %s", port)
 
-	t.Logf("Checking for UDP port is running on %v", udpPort)
-	udpListener, err := net.Listen("udp", fmt.Sprintf(":%s", udpPort))
+	t.Logf("Checking for UDP port is running on %v", port)
+	udpListener, err := net.Listen("udp", fmt.Sprintf(":%s", port))
 	if udpListener != nil {
 		defer udpListener.Close()
 	}
-	require.Errorf(t, err, "host-resolver is not listening on UDP port %s", udpPort)
+	require.Errorf(t, err, "host-resolver is not listening on UDP port %s", port)
 
 	output := netstat(t)
 	require.Contains(t, string(output), fmt.Sprintf("%v/host-resolver", cmd.Process.Pid), "Expected the same Pid")
 }
 
-func TestQueryTCP(t *testing.T) {
-	t.Logf("Checking for TCP port on %s", tcpPort)
-	cmd := run(t, "127.0.0.1", tcpPort, udpPort, "host.rd.test=111.111.111.111,host2.rd.test=222.222.222.222")
+func TestQueryStaticHosts(t *testing.T) {
+	port := randomPort()
+	cmd := runHostResovler(t, []string{"-a", "127.0.0.1", "-t", port, "-u", port, "-c", "host.rd.test=111.111.111.111,host2.rd.test=222.222.222.222"})
```

Comment:
It would be good to use _different_ ports for TCP & UDP, to ensure that we're not accidentally using the other.